app/modules/transcribe/overlay/keypad-slide.directive.js

```diff
@@ -48,6 +47,25 @@ function keypadSlide(hotkeys, overlayConfig) {
             }
             textarea.caret(startTag.length + text.length);
         }
+
+        function preLoad() {
+            var deferred = $q.defer();
+            var imageArray = [];
+            for (var i = 0; i < $scope.abbreviations.length; i++) {
+                imageArray[i] = new Image();
+                imageArray[i].src = $scope.abbreviations[i].imgPath;
+            }
+            imageArray.forEach.onload = function () {
```

Comment:
I don't think this is going to work in the way you intended - this will resolve the promise as soon as an image is loaded, not all of them.

If you want to make sure they're all loaded, you'll need to use `$q.all()`